sonic-chassisd/scripts/chassisd

```diff
@@ -649,19 +955,27 @@ class ChassisdDaemon(daemon_base.DaemonBase):
                                               default=INVALID_SLOT)
         supervisor_slot = try_get(platform_chassis.get_supervisor_slot,
                                                       default=INVALID_SLOT)
-        
+
         # Check if module list is populated
-        self.module_updater = ModuleUpdater(SYSLOG_IDENTIFIER, platform_chassis, my_slot, supervisor_slot)
+        self.smartswitch = platform_chassis.is_smartswitch()
+        self.log_info("smartswitch: {}".format(self.smartswitch))
+        if self.smartswitch:
+            self.module_updater = SmartSwitchModuleUpdater(SYSLOG_IDENTIFIER, platform_chassis)
+        else:
+            self.module_updater = ModuleUpdater(SYSLOG_IDENTIFIER, platform_chassis, my_slot, supervisor_slot)
         self.module_updater.modules_num_update()
 
-
-        if ((self.module_updater.my_slot == INVALID_SLOT) or
-                (self.module_updater.supervisor_slot == INVALID_SLOT)):
-            self.log_error("Chassisd not supported for this platform")
-            sys.exit(CHASSIS_NOT_SUPPORTED)
+        if not self.smartswitch:
+            if ((self.module_updater.my_slot == INVALID_SLOT) or
+                    (self.module_updater.supervisor_slot == INVALID_SLOT)):
+                self.log_error("Chassisd not supported for this platform")
+                sys.exit(CHASSIS_NOT_SUPPORTED)
 
         # Start configuration manager task on supervisor module
-        if self.module_updater.supervisor_slot == self.module_updater.my_slot:
+        if self.smartswitch:
```

Comment:
@rameshraghupathy comment above is incorrect

Reply:
@prgeor Fixed